PublicMapping/district-builder-2
# Containerize "manage" CLI project

## Overview

Containerizes the "manage" CLI project, since it's got to the point where we now have some more complex dependencies that need to be setup (e.g. `tippecanoe`), and we'll soon need database access.

### Demo

![db-containerize-cli](https://user-images.githubusercontent.com/6386/79140165-8d1f9100-7d85-11ea-82b8-34764c22bad3.png)

### Notes

 * This does not add database access, since that will be coming in the following task, but some DB/AWS credentials have been added to the environment in the `docker-compose` file in preparation.
 * The `.prettierrc` file in the root wasn't making its way to the containers, which was causing formatting to behave differently inside and outside the container. Because the config is trivially small, I copied the file to server/manage, rather than complicating the Docker setup. If the `prettier` config ever becomes more complicated, we may want to unify the files.
 * I added a `.gitignore`d directory, `data`, as a place that can be used for storing input/output of the processing command.
 * I added `NODE_OPTIONS=--max-old-space-size=14336` to the `docker-compose` environment vars. 14336MB is the maximum we've needed so far (with California), and there doesn't appear to be any harm in having this set even for a smaller processing job, since it doesn't pre-allocate memory, and is more of a maximum that it's allowed to grow to.
 * @hectcastro: Added you as a reviewer, since you requested Ops involvement on this one. Feel free to reassign as appropriate.

## Testing Instructions

-  `./scripts/update`
- `./scripts/manage`
- You should see command output similar to the above screenshot
- That should be good enough for testing. But if you'd also like to see `tippecanoe` in action, you can do the following to process a GeoJSON file:
  - `mkdir src/manage/data/output-de`
  - Copy `DE.geojson` to `src/manage/data/`
  - `./scripts/manage process-geojson data/DE.geojson -o data/output-de -n 12,4,4 -x 12,12,12 -l block,tract,county`
  - Verify the expected files (topojson, static data, and vector tiles) have been created under `output-de`.

Closes #66
